### DISCUSSION TITLE
Command Line Data Curation Homework

### DISCUSSION THREAD

**Discussion by ZoeLeBlanc** [CultureAsData-UIUC/is310-fall-2024#8] [Assignments]
Please post your Command Line Data Curation Homework here. Details on the assignment are here [https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/creating-curating-humanities-data/04-virtual-environments#homework-command-line-data-curation](https://cultureasdata-uiuc.github.io/is310-fall-2024/materials/creating-curating-humanities-data/04-virtual-environments#homework-command-line-data-curation). Please remember the assignment has two parts.

**Comment by yosefstifanos** [comment]
https://github.com/yosefstifanos/is310-coding-assignments/tree/main/command-line-data-curation

**Reply by kohtas2** [reply]
![screenshot](https://github.com/user-attachments/assets/3ba1e9c4-3ab2-4052-9653-3fec83b8072c)
It worked, but this was not saved to a file!

**Reply by theajk2** [reply]
not sure why I chose to do this one considering I know nothing about sports lol but here
<img width="361" alt="Screenshot 2024-10-14 at 7 53 21 PM" src="https://github.com/user-attachments/assets/b59a70df-237b-4619-98bc-d5e537fedfc6">

**Reply by ZoeLeBlanc** [reply]
Hi @yosefstifanos ! This code looks great but you don't have the data saving to a local file. If you make that update, I'm happy to give you full marks 👍

**Reply by yosefstifanos** [reply]
Hi there. Sorry for not seeing this. I haven't been able to get notifications from GitHub discussions. Maybe because of the email setting where it auto-emails every time someone posts a link or a submission? Idk., but I have fixed this issue now. I hope to receive full marks. Sorry for this again.

**Comment by haydnelm** [comment]
https://github.com/haydnelm/is310-coding-assignments/tree/main/Command-Line-Data-Curation

**Reply by ryanpearlman1** [reply]
[pokemonData.txt](https://github.com/user-attachments/files/17235842/pokemonData.txt)
It would be cool if you could input multiple types for each Pokemon, since some are more than one type!

**Reply by OctaneCodes** [reply]
![Screenshot 2024-10-09 110855](https://github.com/user-attachments/assets/8a32baf5-22fb-48c0-afdf-7019756f2972)
would be cool if you could keep adding pokemon until you didnt want to anymore and not just one instead of having to run the program for each individual entry.

**Reply by ZoeLeBlanc** [reply]
Well done @haydnelm ! You're one of the few that validated the data being entered and also appreciate the helpful `README.md` 🥳 👍 !

**Comment by esr27** [comment]
https://github.com/esr27/is310-coding-assignments/blob/main/cli_data_entry.py

**Reply by ryanpearlman1** [reply]
<img width="207" alt="Screenshot 2024-10-02 at 3 55 56 PM" src="https://github.com/user-attachments/assets/af67a870-c86e-4f7f-98b3-c6d3f615b575">
Here's my dream cars, but I couldn't get them to save to a file.

**Reply by kohtas2** [reply]
![screenshot](https://github.com/user-attachments/assets/bae00ac2-d49f-4226-9cdd-7a400e45bd71)
It worked, but this was not saved to file!

**Reply by elee33** [reply]
<img width="444" alt="Untitled 5" src="https://github.com/user-attachments/assets/748c6d4a-d941-424a-863a-786ecdad3b98">

**Reply by ZoeLeBlanc** [reply]
Hi @esr27 ! This code looks great but you don't have the data saving to a local file. If you make that update, I'm happy to give you full marks 👍

**Comment by ryanpearlman1** [comment]
https://github.com/ryanpearlman1/is310-coding-assignments/tree/main/python-libraries

**Reply by yosefstifanos** [reply]
It worked, but file won't let me attach when replying.

prequels**
<img width="453" alt="image" src="https://github.com/user-attachments/assets/aa34339d-12e6-4322-b224-97d62b1363d8">

**Reply by yosefstifanos** [reply]
also a note; add an option for prequels as well.

**Reply by Cindy1226** [reply]
<img width="210" alt="Screen Shot 2024-10-09 at 8 43 49 PM" src="https://github.com/user-attachments/assets/b62c590d-bec5-4879-9393-31040beaa5bd">

<img width="647" alt="Screen Shot 2024-10-09 at 8 45 04 PM" src="https://github.com/user-attachments/assets/85869fe3-da93-441f-9eab-42ac4f56d5e3">
The code worked great and the data was stored in the.html file

**Comment by kohtas2** [comment]
https://github.com/kohtas2/is310-coding-assignments/tree/main/command-line-data-curation

**Reply by tinayang922** [reply]
It worked, and the last row is what I got. 
<img width="224" alt="Screenshot 2024-10-08 at 9 27 53 AM" src="https://github.com/user-attachments/assets/46ceb9a7-e43c-4461-bb96-800819a6e01f">

**Reply by JBrown-0** [reply]
[arsenal_results.csv](https://github.com/user-attachments/files/17319381/arsenal_results.csv)
Everything worked and here's my results. Big fan of premier league as well.
<img width="548" alt="Screenshot 2024-10-09 at 5 42 07 PM" src="https://github.com/user-attachments/assets/88393d5f-1f4e-409a-ad67-c1a52fd6697d">

[thearsenal_results.csv](https://github.com/user-attachments/files/17319472/thearsenal_results.csv)

**Reply by SamuelSokolovsky** [reply]
<img width="240" alt="Screenshot 2024-10-09 at 9 33 28 PM" src="https://github.com/user-attachments/assets/1c80492d-af34-44a3-9839-9aec20822b29">
added the bottom 2. works very well

**Reply by ZoeLeBlanc** [reply]
Well done @kohtas2 and especially appreciate your helpful `README.md` 👍 !

**Comment by tinayang922** [comment]
https://github.com/tinayang922/is310-coding-assignments/tree/main/python-libraries

**Reply by averyWebster** [reply]
<img width="469" alt="Screenshot 2024-10-08 at 1 56 23 PM" src="https://github.com/user-attachments/assets/65d8786f-bef6-42b2-88e2-60b287f2cd19">
Program worked really well! This is the csv.

**Comment by averyWebster** [comment]
https://github.com/averyWebster/is310-coding-assignments/tree/main/python-libraries

**Reply by AlanK3** [reply]
![image](https://github.com/user-attachments/assets/0a756e65-36a0-4896-beba-2a808a74bfbc)
For some reason, it only worked fully within the VSCode terminal, but crashed in my Windows 11 terminal when I tried to save the food log csv. Regardless, a neat idea

**Reply by ZoeLeBlanc** [reply]
Hi @averyWebster ! This looks great. One issue I noticed in your repo though is that you pushed up your virtual environment in your `python-libraries` folder. I would highly recommend removing it from your remote GitHub repository since it can cause a lot of git issues over time. 

To prevent these issues, you need to add the name of your environment `venv/` to your `.gitignore`, though you might also need to add the `python-libraries/venv/` path since it is nested. However, I would highly recommend deleting this venv environment and recreating it in your root of the your `is310-coding-assignments` folder. That way you can access it for all your Python homework. I realize it's annoying to have to create a fresh environment from scratch but it's far easier than trying to remove it from your git history.

Happy to help if you need assistance but overall great work 👍

**Comment by AlanK3** [comment]
https://github.com/AlanK3/is310-coding-assignments/tree/main/command-line-data-curation

**Reply by GloriaLiu930** [reply]
<img width="485" alt="image" src="https://github.com/user-attachments/assets/d364134c-531d-41df-bb0d-b904fe72729c">
<img width="316" alt="image" src="https://github.com/user-attachments/assets/7da81b55-9dd8-4ee1-a67e-5b36d4bbc0b1">

It works very well! These are the process and the csv.

**Reply by ZoeLeBlanc** [reply]
Hi @AlanK3 ! This looks great. One issue I noticed in your repo though is that you pushed up your virtual environment. I would highly recommend removing it from your remote GitHub repository since it can cause a lot of git issues over time. To do that you'll need to do the following steps:

1. Create a `.gitignore` file in the root of your `is310-coding-assignments folder`
2. Add the name of your environment `IS310venv/` to your `.gitignore` file
3. You then might need to delete the virtual environment and recreate it so it is no longer tracked by git. I realize it's annoying to have to create a fresh environment from scratch but it's far easier than trying to remove it from your git history.

Happy to help if you need assistance but overall great work, and especially impressed with your `README.md` as well as your code 👍

**Comment by GloriaLiu930** [comment]
https://github.com/GloriaLiu930/is310-coding-assignments/tree/main/command-line-data-curation

**Reply by ahmadsawwan** [reply]
<img width="671" alt="Screen Shot 2024-10-09 at 10 06 02 AM" src="https://github.com/user-attachments/assets/b27a836a-8007-49db-9949-7671d22a0081">
Nice! it worked and saved it to a file.

**Reply by ZoeLeBlanc** [reply]
Hi @GloriaLiu930 ! When I try to run your code I get the following error:

<img width="1084" alt="Screenshot 2024-10-21 at 5 20 07 PM" src="https://github.com/user-attachments/assets/c0b06c0a-3a88-4241-b8dd-799ddeea82bd">

It's not a a huge error and I think you should be able to fix it with this error message. If you do fix it, I'll be happy to give you full marks 👍 !

@ahmadsawwan I believe you may have tried out @tinayang922 's cli data entry script?

**Reply by GloriaLiu930** [reply]
@ZoeLeBlanc 
Hi Prof. Blanc,
Appreciate you pointing it out and I already fixed it. I lost my original repo since some mistake push replace, and I have no choice but force push my local back up to GitHub otherwise I'll lose all of the previous files. Thus there might be some small error in the previous assignments =(
Thanks,
Gloria

**Comment by ahmadsawwan** [comment]
https://github.com/ahmadsawwan/ahmadsawwan-is310-coding-assignments/blob/main/cli_data_entry.py

**Reply by shoirathi** [reply]
![Screenshot 2024-10-09 105739](https://github.com/user-attachments/assets/1bbef492-6f1f-4d93-8984-22d706438a13)

**Reply by shoirathi** [reply]
This code worked really well and I was able to store my data in the .json file.

**Reply by jasonmorris8** [reply]
<img width="511" alt="Screenshot 2024-10-09 at 5 22 20 PM" src="https://github.com/user-attachments/assets/ffaae6fd-b40f-42d7-af5a-a5d72a2da30e">
The code worked great and the data was stored in the.json file.

**Reply by elee33** [reply]
![image](https://github.com/user-attachments/assets/1e698bea-6b10-46da-82d2-f277c7417d40)
Code worked! Was able to store into.json.

**Comment by shoirathi** [comment]
https://github.com/shoirathi/is310-coding-assignments/blob/main/python-libraries/cli_data_entry.py

**Reply by LucasWashor** [reply]
<img width="710" alt="IS310-DataCuration-Screenshot" src="https://github.com/user-attachments/assets/6c5ea013-d7bc-46e6-bbaf-8c049495e391">
The code worked great and created a csv file, heres the players I added to your list. Great job!

**Comment by OctaneCodes** [comment]
https://github.com/OctaneCodes/is310-coding-assignments/tree/main/python-libraries

**Reply by jasonmorris8** [reply]
<img width="610" alt="Screenshot 2024-10-09 at 5 26 03 PM" src="https://github.com/user-attachments/assets/f45123ba-b033-42ea-8942-cb85b0396ecb">
The code worked great and the data was saved in a .txt file.

**Comment by theajk2** [comment]
https://github.com/theajk2/is310-coding-assignments/tree/main/command-line-data-curation

**Reply by ZoeLeBlanc** [reply]
Hi @theajk2 ! This is a good start but I got an error when I try to run your code:

<img width="943" alt="Screenshot 2024-10-21 at 9 34 51 PM" src="https://github.com/user-attachments/assets/90b44057-5ca8-4fd6-a4b3-d36179588ec0">

Happy to give you full marks if you fix the issue! Also happy to help if you get stuck 👍

**Reply by theajk2** [reply]
@ZoeLeBlanc I have properly imported rich and fixed the table variable for when I try to create rows and columns. Did that fix it on your end?

**Reply by ZoeLeBlanc** [reply]
@theajk2 still getting errors? Is the code working when you run it?

<img width="907" alt="Screenshot 2024-10-22 at 10 15 34 AM" src="https://github.com/user-attachments/assets/ac65b8da-7ac8-4498-b990-fa701785dd52">

**Reply by theajk2** [reply]
@ZoeLeBlanc I've gotten. that error too but figured it was just a glitch because I cannot for the life of me find the line that is throwing the error. The terminal says it's 405 and I've manually searched for the '_index=len(self.columns) line and I cannot find it anywhere.

**Comment by JBrown-0** [comment]
https://github.com/JBrown-0/is310-coding-assignments/tree/main/python-libraries

**Reply by frangara** [reply]
![image](https://github.com/user-attachments/assets/d4988b79-3d93-4b27-9a6f-c2bdf5fa8e3d)

**Reply by ZoeLeBlanc** [reply]
Well done @JBrown-0 and especially appreciate your helpful `README.md` 👍

**Comment by jasonmorris8** [comment]
https://github.com/jasonmorris8/is310-coding-assignments/tree/main/python-libraries

**Reply by ZoeLeBlanc** [reply]
Hi @jasonmorris8 ! Overall this is a good work but when I ran your code I got the following error:

<img width="1005" alt="Screenshot 2024-10-21 at 7 41 19 PM" src="https://github.com/user-attachments/assets/4ca1ec83-d7b5-41d2-91db-d8d6d3ce1951">

I also noticed that your `README.md` has a set of instructions that doesn't work with your current code (and also all the text is wrapped in a docstring for some reason?). 

If you fix your code, I'm happy to give you full marks 👍

**Reply by jasonmorris8** [reply]
https://github.com/jasonmorris8/is310-coding-assignments/tree/main/python-libraries The issue should be fixed now, I redid most of the code.

**Comment by elee33** [comment]
https://github.com/elee33/Command-Line-Data-Curation

**Reply by chuzesta** [reply]
[favorite_books.json](https://github.com/user-attachments/files/17322174/favorite_books.json)
<img width="405" alt="image" src="https://github.com/user-attachments/assets/ae65dec6-ca47-42e1-a9cc-266346dc6ba6">

**Reply by ZoeLeBlanc** [reply]
Hi @elee33 ! This code looks good but I'm a bit confused. The link above is to a totally separate repository? However, it looks like you have the code in your `is310-coding-assignments` also? If you could just have the code in your `is310-coding-assignments` repo that would be ideal. Thanks!

**Reply by elee33** [reply]
this should work: https://github.com/elee33/is310-coding-assignments/blob/main/cli_data_entry.py

**Comment by Cindy1226** [comment]
https://github.com/Cindy1226/is310-coding-assignments/tree/main/python-libraries

**Reply by divyaj5** [reply]
Hi! Here is the data I was able to add:
<img width="414" alt="image" src="https://github.com/user-attachments/assets/52bc58fa-2311-4b5d-85bf-ff74ebc1f18d">
<img width="647" alt="image" src="https://github.com/user-attachments/assets/c3ea4ea6-71e7-43c7-9dc4-32a303d9f394">
Program worked great! Thank you and good job.

**Reply by ZoeLeBlanc** [reply]
Well done @Cindy1226 and especially impressed that you included a helpful `README.md` 👍

**Comment by SamuelSokolovsky** [comment]
https://github.com/SamuelSokolovsky/is310-coding-assignments/blob/main/python_libraries/cli_data_entry.py

**Reply by ZoeLeBlanc** [reply]
Good work @SamuelSokolovsky 👍 !

<img width="623" alt="Screenshot 2024-10-21 at 9 03 35 PM" src="https://github.com/user-attachments/assets/58584a22-d33d-4034-83db-009d0a97b7e0">

**Comment by henRotunno** [comment]
https://github.com/henRotunno/is310-coding-assignments/tree/main/python-libraries

**Reply by ZoeLeBlanc** [reply]
Hi @henRotunno ! I tried out your code and it works great. 

<img width="673" alt="Screenshot 2024-10-21 at 7 32 12 PM" src="https://github.com/user-attachments/assets/b3e90868-6ec3-4ae1-9649-5554255f2a97">

One small thing is that you have a space in your script name `from rich.py` and also the name for the script is a bit confusing. I would recommend renaming it `cli_data_entry.py`. Overall though good work 👍

**Comment by frangara** [comment]
https://github.com/frangara/is310-coding-assignments/tree/main/python-libraries

**Reply by Rebe-Rod** [reply]
![Screenshot 2024-10-09 233123](https://github.com/user-attachments/assets/fc6973cd-e7c0-4287-81f5-ff17a094a5d9)
 This was very well done!

**Reply by chuzesta** [reply]
<img width="405" alt="image" src="https://github.com/user-attachments/assets/081af75d-6f50-4a27-8d68-73a6974b7415">

**Comment by LucasWashor** [comment]
https://github.com/LucasWashor/is310-coding-assignments/tree/main/python-libraries

**Reply by officalcharles** [reply]
![Screenshot 2024-10-13 at 6 46 36 PM](https://github.com/user-attachments/assets/4284aa92-20d4-4a2d-aa3f-3f0863dc1a0f)
Great Job! Works well

**Reply by ZoeLeBlanc** [reply]
Hi @LucasWashor ! This is great work but just noticed you pushed up your virtual environment, which we try to avoid adding to our GitHub repositories.

You also have the virtual environment in your `python-libraries` folder. I would recommend deleting your current `.venv`. Then making a new virtual environment, but this time in the root of your `is310-coding-assignments`. Then you should create a `.gitignore` and add the name of the environment to the `.gitignore`.

I'm happy to help if you get stuck and overall well done, and especially appreciate your helpful `README.md` 👍

**Comment by Rebe-Rod** [comment]
https://github.com/Rebe-Rod/is310-coding-assingments/tree/main/python-libraries

**Reply by henRotunno** [reply]
<img width="604" alt="Screenshot 2024-10-10 at 4 10 50 PM" src="https://github.com/user-attachments/assets/e445f6e1-6feb-45b9-9214-276df03f1f43">
shoutout cross country club for the steps

**Comment by chuzesta** [comment]
https://github.com/chuzesta/is310-coding-assignments/tree/main/command-line-data-curation

**Reply by henRotunno** [reply]
<img width="624" alt="Screenshot 2024-10-10 at 4 08 34 PM" src="https://github.com/user-attachments/assets/c7857fbe-dcaf-4f9c-8853-590c6da01935">

Nice!

**Reply by ZoeLeBlanc** [reply]
Hi @chuzesta ! Overall your good works great but two issues I noticed. 

First you're saving the console.capture object which produces fairly difficult data to reuse:

<img width="926" alt="Screenshot 2024-10-21 at 7 54 10 PM" src="https://github.com/user-attachments/assets/9d1aa0ed-a37d-4a3b-8b28-a1689b2449f7">

I would recommend instead saving the actual data to whatever format your prefer.

Second, I noticed you pushed up both a `.venv` and `.vscode` in your `command-line-data-curation` folder. I would recommend not pushing those files to GitHub because it can cause a number of issues with git over time. All you need to do is add that environment and vscode files to your `.gitignore` and/or simply delete the `.venv` and just use the `is310-env` you're referencing in your `.gitignore`. Happy to help if you need assistance and again good work 👍

**Comment by officalcharles** [comment]
https://github.com/officalcharles/is310-coding-assignments/tree/master/python-libraries

**Reply by lhayedimarucut** [reply]
<img width="352" alt="Screenshot 2024-10-14 at 6 24 04 PM" src="https://github.com/user-attachments/assets/a16f9f16-3ad3-4936-aa89-23093de0d379">
Was able to give me the css. Great job!

**Reply by ZoeLeBlanc** [reply]
Hi @officalcharles  ! This looks great. One issue I noticed in your repo though is that you pushed up an `.idea` folder in your `python-libraries` folder. From my understanding, an `.idea` folder comes from PyCharm. Generally, these type of environment folders are not ideal to have in your remote GitHub repository since it can cause a lot of git issues over time. So would recommend trying to remove it from your GitHub and add that folder to your `.gitignore`:

1. Create a `.gitignore` file in the root of your `is310-coding-assignments folder`
2. Add the `python-libraries/.idea` folder to your `.gitignore`.
3. You then might need to delete the `.idea` folder to have it fully be removed from your GitHub repository. I believe that PyCharm generates it automatically so that shouldn't cause issues but would recommend taking a look at this Stack Overflow discussion to determine how you want to proceed [https://stackoverflow.com/questions/24516814/should-i-ignore-the-idea-folder-when-using-pycharm-with-git](https://stackoverflow.com/questions/24516814/should-i-ignore-the-idea-folder-when-using-pycharm-with-git).

Happy to help if you need assistance but overall great work 👍

**Comment by lhayedimarucut** [comment]
https://github.com/lhayedimarucut/is310-coding-assignments/blob/main/python-libraries/cli_data_entry.py 

Sorry was late to submit due to issue pushing it into git

**Reply by ZoeLeBlanc** [reply]
Hi @lhayedimarucut ! Overall this is good work but I ran into an error when I tried running your code:

<img width="949" alt="Screenshot 2024-10-21 at 8 15 10 PM" src="https://github.com/user-attachments/assets/45001121-d0a0-4b6e-8fd2-46a2b3575cc3">

Happy to give you full marks if you fix the issue and also to provide assistance if you are stuck 👍

**Comment by adg77** [comment]
https://github.com/adg77/is310-coding-assignments/tree/main/python-libraries

**Reply by divyaj5** [reply]
Added the last row to your dataset: 
<img width="749" alt="image" src="https://github.com/user-attachments/assets/855b53de-4e76-4760-af75-573af1ad835c">
Really liked how you included box office earnings as an attribute as thats not a quality of movies that most people think about! Great Job.

**Comment by divyaj5** [comment]
https://github.com/divyaj5/is310-coding-assignments/tree/master/command-line-data-curation

**Reply by ZoeLeBlanc** [reply]
Well done @divyaj5 and especially impressed with your helpful `README.md` 👍 !

**Comment by AG-Samaniego** [comment]
(Still a work in progress, will leave it here for now and update thread once it works): https://github.com/AG-Samaniego/is310-coding-assignments/tree/main/python-libraries

**Reply by yukthas2** [reply]
<img width="554" alt="Screenshot 2024-11-18 at 12 05 08 AM" src="https://github.com/user-attachments/assets/30d6053d-e0cd-4016-97fc-58f68a4722ec">

**Comment by yukthas2** [comment]
https://github.com/yukthas2/is310-coding-assignments/tree/main/python-libraries
